json-schema-processor/src/test/groovy/io/micronaut/jsonschema/visitor/JsonSchemaVisitorSpec.groovy

```diff
@@ -0,0 +1,357 @@
+package io.micronaut.jsonschema.visitor
+
+import io.micronaut.jsonschema.visitor.model.Schema
+
+
+class JsonSchemaVisitorSpec extends AbstractJsonSchemaSpec {
+
+    void "simple record schema"() {
+        given:
+        def schema = buildJsonSchema('test.Salamander', 'salamander', """
+        package test;
+
+        import io.micronaut.jsonschema.JsonSchema;
+        import java.util.*;
+
+        @JsonSchema
+        public record Salamander(
+                String name,
+                int age,
+                Color color,
+                List<String> environments,
+                Map<String, List<String>> complexMap
+        ) {
+        }
+
+        enum Color {
+            RED,
+            GREEN,
+            BLUE
+        }
+""")
+
+        expect:
+        schema.title == "Salamander"
+        schema.properties['name'].type == [Schema.Type.STRING]
+        schema.properties['age'].type == [Schema.Type.INTEGER]
+        schema.properties['color'].type == [Schema.Type.STRING]
+        schema.properties['color'].enumValues == ["RED", "GREEN", "BLUE"]
```

Comment:
@sdelamo  We support generation of enums. See the `Color` enum in this test.

But this is a little complicated, as we cannot know how the user wants to encode the enum if they use `@JsonValue`, for example.